src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -504,111 +504,98 @@ export class StudyViewPageStore {
-                        const patientKeyToData = _.keyBy(data, (d:ClinicalData)=>d.uniquePatientKey);
-                        for (const sample of selectedSamples) {
-                            const datum = patientKeyToData[sample.uniquePatientKey];
-                            const value = datum ? datum.value : "NA";
-                            const lcValue = value.toLowerCase();
-                            lcValueToValue[lcValue] = lcValueToValue[lcValue] || value;
-                            lcValueToSampleIdentifiers[lcValue] = lcValueToSampleIdentifiers[lcValue] || [];
-                            lcValueToSampleIdentifiers[lcValue].push({
-                                sampleId: sample.sampleId,
-                                studyId: sample.studyId
+        // for now, the only one possible is cancer studies
+        return new Promise<string>(resolve => {
+            onMobxPromise<any>(promises,
+                async (selectedSamples: Sample[], clinicalDataList?: ClinicalData[]) => {
+
+                    let data: CustomChartIdentifierWithValue[] = [];
+
+                    if (this.isCustomChart(chartMeta.uniqueKey)) {
+                        if (clinicalDataList === undefined) {
+                            //combine studyId and sampleId to create sample unique key
+                            const sampleKeyToData = _.keyBy(this._customChartsSelectedCases.get(chartMeta.uniqueKey), d => d.studyId + d.sampleId);
+                            data = selectedSamples.map(sample => {
+                                const datum = sampleKeyToData[sample.studyId + sample.sampleId];
+                                return {
+                                    sampleId: sample.sampleId,
+                                    studyId: sample.studyId,
+                                    value: datum ? datum.value : "NA",
+                                } as CustomChartIdentifierWithValue
                             });
+                        } else {
+                            const selectedSampleByKey = _.keyBy(selectedSamples, (d: Sample) => d.uniqueSampleKey);
+                            data = _.filter(clinicalDataList, d => selectedSampleByKey[d.uniqueSampleKey] !== undefined);
                         }
                     } else {
-                        lcValueToSampleIdentifiers = _.groupBy(data, (d:ClinicalData)=>{
-                            const value = d.value;
-                            const lcValue = value.toLowerCase();
-                            lcValueToValue[lcValue] = lcValueToValue[lcValue] || value;
-                            return lcValue;
+
+                        // get clinical data for the given attribute
+                        const isPatientAttribute = (chartMeta.clinicalAttribute && chartMeta.clinicalAttribute.patientAttribute) || false;
+                        const entityIdKey = (isPatientAttribute ? "patientId" : "sampleId")
```

Comment:
This should be in the util as well

Reply:
haven't found any util for this. anyways this is going to be removed in study-view url cleanup pr